mlflow/gateway/config.py

```diff
@@ -0,0 +1,274 @@
+import pathlib
+from enum import Enum
+import json
+import logging
+import os
+from pathlib import Path
+from pydantic import BaseModel, validator, parse_obj_as, Extra, ValidationError
+from pydantic.json import pydantic_encoder
+from typing import Optional, Union, List, Dict, Any
+import yaml
+
+from mlflow.exceptions import MlflowException
+from mlflow.gateway.constants import PROVIDERS
+from mlflow.gateway.utils import is_valid_endpoint_name, check_configuration_route_name_collisions
+
+
+_logger = logging.getLogger(__name__)
+
+
+class Provider(str, Enum):
+    CUSTOM = "custom"
+    OPENAI = "openai"
+    ANTHROPIC = "anthropic"
+    DATABRICKS_SERVING_ENDPOINT = "databricks_serving_endpoint"
+    MLFLOW = "mlflow"
+
+
+class RouteType(str, Enum):
+    CUSTOM = "custom"
+    LLM_V1_COMPLETIONS = "llm/v1/completions"
+    LLM_V1_CHAT = "llm/v1/chat"
```

Comment:
@dbczumar @BenWilson2 Are we missing `embeddings` here?

Reply:
Yes, we should adding embeddings. Embeddings are P0, chat is P1

Reply:
Filed https://github.com/mlflow/mlflow/pull/8711